controllers/auth.js

```diff
@@ -9,11 +25,22 @@ const authService = require("../services/authService");
  * @param res {Object} - Express response object
  * @param next {Function} - Express middleware function
  */
-const githubAuth = (req, res, next) => {
+const githubAuthCallback = (req, res, next) => {
   let userData;
   const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl"));
-  let authRedirectionUrl = req.query.state ?? rdsUiUrl;
-
+  let authRedirectionUrl = rdsUiUrl;
+  if ("state" in req.query) {
+    try {
+      const redirectUrl = new URL(req.query.state);
+      if (redirectUrl.hostname.endsWith(REALDEVSQUAD_HOSTNAME)) {
```

Comment:
I do not think using `endWith()` is a good idea.
Because if is pass `state = https://rrealdevsquad.com`(notice the extra r at the beginning) the condition will evaluate to true